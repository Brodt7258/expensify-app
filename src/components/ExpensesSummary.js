import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = ({ quantity, total }) => {
  return (
    !!quantity 
    &&
    <div>
      <p>Viewing {quantity} expense{quantity > 1 && 's'} totaling {numeral(total / 100).format('$0,0.00')}</p>
    </div>
  );
};

const mapStateToProps = (state) => {
  const visExpenses = selectExpenses(state.expenses, state.filters);
  return {
    quantity: visExpenses.length,
    total: selectExpensesTotal(visExpenses)
  };
};

export default connect(mapStateToProps)(ExpensesSummary);