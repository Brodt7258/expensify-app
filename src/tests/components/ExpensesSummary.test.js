import React from 'react';
import { shallow } from 'enzyme';
import expenses from '../fixtures/expenses';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import selectExpensesTotal from '../../selectors/expenses-total';

test('should correctly render ExpensesSummary with a single expense', () => {
  const quantity = 1;
  const total = 100;
  const wrapper = shallow(<ExpensesSummary quantity={quantity} total={total}/>);
  expect(wrapper).toMatchSnapshot();
});

test('should correctly render ExpensesSummary with multiple expenses', () => {
  const quantity = 3;
  const total = 415300;
  const wrapper = shallow(<ExpensesSummary quantity={quantity} total={total}/>);
  expect(wrapper).toMatchSnapshot();
});

test('should correctly render nothing with no expenses', () => {
  const quantity = 0;
  const total = 0;
  const wrapper = shallow(<ExpensesSummary quantity={quantity} total={total}/>);
  expect(wrapper).toMatchSnapshot();
});