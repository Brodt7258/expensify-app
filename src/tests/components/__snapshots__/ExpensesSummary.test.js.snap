// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpensesSummary with a single expense 1`] = `
<div>
  <p>
    Viewing 
    1
     expense
     totaling 
    $1.00
  </p>
</div>
`;

exports[`should correctly render ExpensesSummary with multiple expenses 1`] = `
<div>
  <p>
    Viewing 
    3
     expense
    s
     totaling 
    $4,153.00
  </p>
</div>
`;

exports[`should correctly render nothing with no expenses 1`] = `undefined`;
